NSubstitute.DbConnection/NSubstitute.DbConnection/MockQuery.cs

```diff
@@ -24,22 +24,70 @@ public IMockQueryBuilder WithParameter(string name, object value)
         {
             if (Parameters == null)
             {
-                Parameters = new Dictionary<string, object>();
+                Parameters = new Dictionary<string, QueryParameter>();
             }
 
-            Parameters[name] = value;
+            Parameters[name] = new QueryParameter(name, value);
             return this;
         }
 
         public IMockQueryBuilder WithParameters(IReadOnlyDictionary<string, object> parameters)
         {
-            Parameters = parameters.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+            if (Parameters == null)
+            {
+                Parameters = new Dictionary<string, QueryParameter>();
+            }
+
+            foreach (var kvp in parameters)
+            {
+                Parameters.Add(kvp.Key, new QueryParameter(kvp.Key, kvp.Value));
+            }
+
             return this;
         }
 
         public IMockQueryBuilder WithParameters(params (string Key, object Value)[] parameters)
         {
-            Parameters = parameters.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
```

Comment:
With two methods becoming four here, and the implementation expanding from duplicating a single line twice to duplicating several lines four times, I wonder can these four implementations be extracted into a helper method that each calls with appropriate args?